sources/packages/api/src/route-controllers/user/user.controller.ts

```diff
@@ -1,13 +1,16 @@
 import { Controller, Get, Patch } from "@nestjs/common";
-import { UserService } from "../../services";
+import { StudentService, UserService } from "../../services";
 import BaseController from "../BaseController";
 import UserSyncInfoDto from "./model/user.dto";
 import { UserToken } from "src/auth/decorators/userToken.decorator";
 import { IUserToken } from "src/auth/userToken.interface";
 
 @Controller("users")
 export class UserController extends BaseController {
-  constructor(private readonly service: UserService) {
+  constructor(
+    private readonly service: UserService,
+    private readonly studentService: StudentService,
```

Comment:
Because this is now so much to do with student and not user per se, i think it makes more sense to have all this code in student.controller.ts

Reply:
I struggled a bit with it also because it is gonna update data on the user(first name, last name, and email) and also on the student (DOB and gender).
We can move it to student, but then I believe that we should move it to student controller and rename the method from synchronizeUserInfo to synchronizeStudentInfo on API endpoint, what do you think?